### PR TITLE
Introduce new variable 'rocksdb_error_on_suboptimal_collation'.

### PR DESCRIPTION
**Pull request by george-lorch**
New variable specifies whether to raise an error or a warning if an index is
created on a char field where the table has a sub-optimal collation (case
insensitive).

Default is TRUE. Variable is global read only.

### CODE REVIEW

File: mysql-test/suite/rocksdb/t/collation.test [hunk 1]
```diff
@@ -29,6 +27,8 @@ DROP TABLE t1;
 
 # cs latin1_bin is allowed
 CREATE TABLE t1 (id varchar(20), value varchar(50), value2 varchar(50), value3 text, primary key (id), index(value, value2)) engine=rocksdb charset latin1 collate latin1_bin;
+# THIS SHOULD FAIL BUT IT DOES NOT
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/collation.test:30]: ?

File: storage/rocksdb/ha_rocksdb.cc [hunk 2]
```diff
@@ -492,6 +492,7 @@ static my_bool rocksdb_large_prefix = 0;
 static my_bool rocksdb_allow_to_start_after_corruption = 0;
 static uint32_t rocksdb_write_policy =
     rocksdb::TxnDBWritePolicy::WRITE_COMMITTED;
+static my_bool rocksdb_error_on_suboptimal_collation = 1;
```
Comment (laurynas-biveinis) [inline storage/rocksdb/ha_rocksdb.cc:495]: s/1/TRUE

File: mysql-test/suite/rocksdb/t/collation.test (hunk 1 above)
Comment (george-lorch) [reply mysql-test/suite/rocksdb/t/collation.test:30]: Added simple test and notation to cover https://github.com/facebook/mysql-5.6/issues/779 which is related/overlaps this test. See also https://jira.percona.com/browse/MYR-223

File: storage/rocksdb/ha_rocksdb.cc (hunk 2 above)
Comment (george-lorch) [reply storage/rocksdb/ha_rocksdb.cc:495]: Not in upstream fb, 0/1 is the standard. We 'fix' it on merge downstream.

**Pull request closed without merging**